sstables/checksum_utils.hh

```diff
@@ -65,6 +65,9 @@ struct zlib_crc32_checksummer {
     }
 
     inline static uint32_t checksum_combine(uint32_t first, uint32_t second, size_t input_len2) {
+        if (input_len2 == 0) {
```

Comment:
Does this fix any user-observable bug?

Reply:
I don't think any production code path in Scylla calls the "reference" zlib crc32 implementation (except on extremely exotic and very unsupported platforms), so probably not?

But this definitely needs some unpacking.

The immediately-problematic thing is Scylla's wrong assumption about an edge case in `crc32_combine()`'s contract (the author (reasonably IMO, given he is not paid to maintain backwards bug-compatibility) does not think the new behavior is a bug and is not going to change it: https://github.com/madler/zlib/issues/659#issuecomment-1146288448).

Sometime Soon Scylla will be linked against zlib 1.2.12 or newer (I hit it already, after updating my build environment to the newly-released Nixpkgs 22.05, but I'm sure it'll appear in dbuild via Fedora within a year at most).  I see 3 options:

1. Change `fast_crc32_combine()` to behave like the new reference
2. Wrap any calls to zlib's `crc32_combine()` such that Scylla's idea of its contract continues to hold
3. Change `checksum_utils_test` to give more valid inputs to `crc32_combine()` in the zero-length case, or stop testing the zero-length case altogether

Option 1 looks like a non-starter, unless we somehow know for sure that no results of calling `fast_crc32_combine()` with zero `len2` and non-zero `crc2` have ever been written out into existing user data (well, I at least don't know that).

Option 3 would be ignoring the problem (either bluntly or subtly).

So I went with option 2 here.

Reply:
The author's comment which says that zero-length crc should always be zero seems legit. Did you encounter a case where this doesn't hold? I still don't see why we need this patch.

Reply:
`boost/checksum_utils_test.test_default_matches_zlib` fails here: https://github.com/scylladb/scylla/blob/605ee74c39b241043d75015d14806ac12f4febb3/test/boost/checksum_utils_test.cc#L25

Reply:
Let's fix the test for len=0 then.